src/test/java/com/hyperwallet/clientsdk/HyperwalletTest.java

```diff
@@ -2892,6 +2892,231 @@ public void testCreateTransferStatusTransition_successful() throws Exception {
         assertThat(apiClientTransition.getCreatedOn(), is(nullValue()));
     }
 
+    //--------------------------------------
+    // Transfer Refund
+    //--------------------------------------
+
+    @Test
+    public void testCreateTransferRefund_noTransferRefund() {
+        Hyperwallet client = new Hyperwallet("test-username", "test-password");
+        try {
+            client.createTransferRefund("transferToken",null);
+            fail("Expect HyperwalletException");
+        } catch (HyperwalletException e) {
+            assertThat(e.getErrorCode(), is(nullValue()));
+            assertThat(e.getResponse(), is(nullValue()));
+            assertThat(e.getErrorMessage(), is(equalTo("Transfer Refund is required")));
+            assertThat(e.getMessage(), is(equalTo("Transfer Refund is required")));
+            assertThat(e.getHyperwalletErrors(), is(nullValue()));
+            assertThat(e.getRelatedResources(), is(nullValue()));
```

Comment:
Same with this, is it possible to extract this to a common re-usable method.

Reply:
done